### PR TITLE
Remove legacy disclosed proof

### PR DESCRIPTION
**Pull request by mirgee**
Signed-off-by: Miroslav Kovar <miroslavkovar@protonmail.com>

### CODE REVIEW

File: libvcx/src/disclosed_proof.rs [hunk 1]
```diff
@@ -273,303 +219,108 @@ pub fn build_rev_states_json(credentials_identifiers: &mut Vec<CredInfo>) -> Vcx
...
-        anoncreds::libindy_prover_get_credentials_for_proof_req(&indy_proof_req)
+            rtn[cred_info.schema_id.to_owned()] = schema_json;
+        }
     }
+    Ok(rtn.to_string())
+}
 
-    pub fn build_schemas_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
-        let mut rtn: Value = json!({});
+pub fn build_cred_def_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
```
Comment (Patrik-Stas) [inline libvcx/src/disclosed_proof.rs:239]: Seems like this is duplication of `build_credential_defs_json` in`proof_utils.rs`

File: libvcx/src/disclosed_proof.rs [hunk 2]
```diff
@@ -273,303 +219,108 @@ pub fn build_rev_states_json(credentials_identifiers: &mut Vec<CredInfo>) -> Vcx
...
-        self.state as u32
-    }
-    fn set_state(&mut self, state: VcxStateType) {
-        trace!("DisclosedProof::set_state >>> state: {:?}", state);
-        self.state = state
-    }
-
-    fn retrieve_credentials(&self) -> VcxResult<String> {
-        trace!("DisclosedProof::set_state >>>");
+pub fn build_schemas_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
```
Comment (Patrik-Stas) [inline libvcx/src/disclosed_proof.rs:222]: Seems like this is duplication of `build_credential_defs_json` in`proof_utils.rs`

File: libvcx/src/disclosed_proof.rs [hunk 3]
```diff
@@ -273,303 +219,108 @@ pub fn build_rev_states_json(credentials_identifiers: &mut Vec<CredInfo>) -> Vcx
...
         }
-        Ok(rtn.to_string())
     }
+    Ok(rtn.to_string())
+}
 
-    pub fn build_cred_def_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
-        let mut rtn: Value = json!({});
-
+pub fn build_requested_credentials_json(credentials_identifiers: &Vec<CredInfo>,
```
Comment (Patrik-Stas) [inline libvcx/src/disclosed_proof.rs:256]: Let's move these static functions to `proof_utils.rs` (or `disclosed_proof_utils.rs` if it make more sense for these functions)

File: libvcx/src/v3/handlers/proof_presentation/prover/prover.rs [hunk 4]
```diff
@@ -174,3 +174,259 @@ impl Prover {
         }
     }
 }
+
+#[cfg(test)]
```
Comment (Patrik-Stas) [inline libvcx/src/v3/handlers/proof_presentation/prover/prover.rs:178]: These tests should be moved to `disclosed_proof.rs` as they are calling `Prover::create` (I suppose you were trying to stay consistent with `cleanup/proof` before we reverted the location of its similar tests)

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]

File: libvcx/src/disclosed_proof.rs (hunk 2 above)
Comment (mirgee) [reply libvcx/src/disclosed_proof.rs:222]: Good catch!

File: libvcx/src/disclosed_proof.rs (hunk 1 above)
Comment (mirgee) [reply libvcx/src/disclosed_proof.rs]: It isn't as discussed, moved

File: libvcx/src/disclosed_proof.rs [hunk 5]
```diff
@@ -787,7 +773,7 @@ mod tests {
             tails_file: None,
             timestamp: None,
         }];
-        assert_eq!(build_cred_def_json(&credential_ids).unwrap_err().kind(), VcxErrorKind::InvalidProofCredentialData);
```
Comment (Patrik-Stas) [inline libvcx/src/disclosed_proof.rs]: I think the tests asserting methods from `proof_utils.rs` should be in that file

File: libvcx/src/disclosed_proof.rs [hunk 6]
```diff
@@ -1238,15 +720,15 @@ mod tests {
             tails_file: None,
             timestamp: None,
         }];
-        assert_eq!(DisclosedProof::build_schemas_json(&credential_ids).unwrap_err().kind(), VcxErrorKind::InvalidSchema);
+        assert_eq!(build_schemas_json_prover(&credential_ids).unwrap_err().kind(), VcxErrorKind::InvalidSchema);
```
Comment (Patrik-Stas) [inline libvcx/src/disclosed_proof.rs:723]: I think the tests asserting methods from `proof_utils.rs` should be in that file

Comment (mirgee) [reply libvcx/src/disclosed_proof.rs:723]: Good point

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**